### PR TITLE
Api

### PR DESCRIPTION
**Pull request by Ru-T**
## Why?
- To allow dev to set up a Rails template for an API only app
## What Changed?

If API only: 
- Remove helpers, views, javascripts, stylesheets
- Edit application_controller
- Create alternate Gemfile with api-only gems and without style gems
- If yes to devise, devise_auth_token instead of devise gem
- If yes to api_only, do not ask for ActiveAdmin or Cucumber/Capybara

### CODE REVIEW

File: files/Gemfile_api_only [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+gem 'rails', '4.2.5'
+# Use postgresql as the database for Active Record
+gem 'pg'
+gem 'sdoc', '~> 0.4.0', group: :doc
+# Use Unicorn as the app server
+gem 'unicorn'
+# Use Taperole for deployment
+gem 'taperole'
+
+# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
```
Comment (albaer) [inline files/Gemfile_api_only:13]: Are these optional and would be uncommented as needed by the devs?

Comment (Ru-T) [inline files/Gemfile_api_only:13]: @albaer Yes, that is the idea. I don't want to enforce gems on people, but they are there to uncomment if you want them. 

I did the same with letter_opener. And jbuilder when it's not api-only.

File: template.rb [hunk 2]
```diff
@@ -8,10 +8,27 @@ def render_file(path)
 end
 
 # -----------------------------
+# API ONLY APP?
+# -----------------------------
+if yes?("Is this an API only app with no front-end or admin interface? (y/n)")
```
Comment (dkniffin) [inline template.rb:13]: We may also want to handle the case where it's an API app, with an admin interface.

File: template.rb [hunk 3]
```diff
@@ -91,46 +105,69 @@ def render_file(path)
 end
 
 # Devise
-if yes?("Add Devise?")
+if yes?("Add Devise or Devise_Auth (if api only app)? (y/n)")
```
Comment (dkniffin) [inline template.rb]: You could probably have this be dynamic, so it'll say "Add devise?" or "Add devise_auth?" based on your previous questions.

File: template.rb [hunk 4]
```diff
@@ -91,46 +105,69 @@ def render_file(path)
...
-if yes?("Add ActiveAdmin?")
-  active_admin = true
-  gsub_file 'Gemfile', /^gem\s+["']devise["'].*$/,''
-  inject_into_file 'Gemfile', after: "gem 'taperole'\n" do <<-RUBY
+unless api_only
+  # ActiveAdmin
+  if yes?("Add ActiveAdmin? (y/n)")
+    active_admin = true
+    gsub_file 'Gemfile', /gem 'devise'/, ""
+    inject_into_file 'Gemfile', after: "gem 'taperole'\n" do <<-RUBY
```
Comment (dkniffin) [inline template.rb:131]: I can't remember. Is tape optional? If so, this will break if someone says no for that.

Comment (Ru-T) [inline template.rb:131]: @dkniffin Currently tape is not optional, but rather included in both versions of the Gemfile. 

Do you think it needs to be optional?

Comment (dkniffin) [inline template.rb:131]: Nah, I think we use it on every new app we make, so it's fine to make it required.

**Comment by dkniffin** [comment]
Other than the above, :+1:

File: template.rb (hunk 2 above)
Comment (Ru-T) [inline template.rb:13]: @dkniffin Yes, I agree. I'll do that in a separate pull request if that's cool.

**Comment by TylerRockwell** [comment]
I welcome the additions. The prep work for an API is quite different. It seems like the template has grown to one giant sequential set of steps. It may be worth it for the sake of readability to break it out into smaller methods. It will not only be easier to follow, but you'll be able to reduce the number of checks to see if api_only is set.

Something like

``` RUBY
# check if user wants api_only

if api_only
  api_install_path
else
  regular_install_path
end
```

File: template.rb (hunk 3 above)
Comment (Ru-T) [inline template.rb]: Good call.

**Comment by Ru-T** [comment]
Thanks @TylerRockwell - cleaning it up once all the options are in there with their intended behavior. Thanks @albaer and @dkniffin!

**Comment by TylerRockwell** [comment]
:+1:

**Pull request merged**